src/Properties/SchemaAggregator.php

```diff
@@ -372,14 +373,17 @@ private function processColumnUpdates(string $tableName, string $argName, array
                             $table->setColumn(new SchemaColumn($columnName, 'set', $nullable, $secondArgArray));
                             break;
 
-                        case 'softdeletestz':
                         case 'timestamptz':
                         case 'timetz':
                         case 'year':
-                        case 'softdeletes':
                             $table->setColumn(new SchemaColumn($columnName, 'string', true));
                             break;
 
+                        case 'softdeletestz':
+                        case 'softdeletes':
+                            $table->setColumn(new SchemaColumn($columnName, '?'.get_class(Date::now()), true));
```

Comment:
I think there is no need to add the `?` there. The 3rd argument controls if the column is nullable or not.

Reply:
I think this PR might need a rethink - the reason the nullable parameter is ignored is because the type is not caught in the switch statement in when setting the readable and writable types later on:

https://github.com/nunomaduro/larastan/blob/6e74c6eeb0b27227c787917afd559a2d22ec241c/src/Properties/ModelPropertyExtension.php#L162-L205

Further to that, there is a bit of work on lines 167:169 above to catch model dates such as created_at and updated_at (it also figures out the correct date class to return there in `getDateClass()`).

So it may be more in line with previous work to update lines 167:169 to see if the model has the `SoftDeletes` trait, call the `getDeletedAtColumn()` method if so and then add it to the array of dates to check the provided column name against?

Reply:
Yes that'd be simpler solution I guess. You can even just do `method_exists` check for `getDeletedAtColumn` I think.